Fakebook.Profile/Fakebook.Profile.DataAccess/ProfileRepository.cs

```diff
@@ -241,6 +241,43 @@ public async Task<IEnumerable<DomainProfile>> GetProfilesByEmailAsync(IEnumerabl
                 .ToList();
         }
 
+        public async Task<IEnumerable<DomainProfile>> GetProfilesByNameAsync(string firstName = null, string lastName = null)
+        {
+            List<EntityProfile> query;
+            if(firstName != null && lastName != null)
+            {
+                query = await _context.EntityProfiles
+                .Include(x => x.Following)
+                    .ThenInclude(x => x.Following)
+                .Include(x => x.Followers)
+                    .ThenInclude(x => x.User)
+                .Where(x => x.FirstName.Contains(firstName) && x.LastName.Contains(lastName))
+                .ToListAsync();
+            }
+            else if(lastName == null)
+            {
+                query = await _context.EntityProfiles
+                .Include(x => x.Following)
+                    .ThenInclude(x => x.Following)
+                .Include(x => x.Followers)
+                    .ThenInclude(x => x.User)
+                .Where(x => x.FirstName.Contains(firstName))
+                .ToListAsync();
+            }
+            else
+            {
+                query = await _context.EntityProfiles
+                .Include(x => x.Following)
+                    .ThenInclude(x => x.Following)
+                .Include(x => x.Followers)
+                    .ThenInclude(x => x.User)
+                .Where(x => x.LastName.Contains(lastName))
+                .ToListAsync();
+            }
```

Comment:
build the first 5 lines of the expression before the condition, and then do `query = query.Where // ...` in the branches, then finish it after the conditions. and dont forget about null firstname case

Reply:
I changed the method so we only have one parameter to search by, this will be the entire contents of the search box so we don't have to deal with ambiguity in cases where a user wants to search by last name or first name only.

Reply:
this way we have to load every single profile just to return maybe 1 or 0 of them. you want SQL to do the filtering.
but we're in a hurry, so if you want to just leave a comment to note the issue, or a GitHub issue, instead of changing it then you can do that. do one of the two